docs/manual/src/interface/objects.md

```diff
@@ -0,0 +1,281 @@
+# Objects
+
+Objects are represented in the Rust world as a struct with an `impl` block containing methods.
+In the foreign-language bindings they would typically correspond to a class.
+
+In the interface definition, objects may have public methods but *must not have public fields*,
+because they are intended to be opaque to the foreign-language code. Like this:
+
+```rust
+#[uniffi::declare_interface]
+mod example {
+    struct TodoList {
```

Comment:
This might make the docs more unwieldy than you like, but to help with the mental model, maybe consider:

* Having the `struct` definition outside of the module, which a comment explaining it need not be inside.
* Another `impl TodoList` block outside the module, explaining that these can be thought of as "uniffi-private"

(edit: now that I get down to record.md I'm not actually sure about this)